### PR TITLE
[BE] feat(13): added confirm email endpoint

### PR DESCRIPTION
**Pull request by dartwlad**
suggested flow:
- user click on the link in the email and enter example.com/confirm?emailVerificationToken=some_token
- FE app makes POST call to /api/auth/confirm with {emailVerificationToken: some_token}
- BE returns user profile with token

### CODE REVIEW

File: packages/server/src/services/email-sending.service.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    }
+  }
+
+  public async sendActivationMessage(userEmail: string, host: string, emailVerificationToken: string): Promise<void> {
+    const link = `${host}${Pages.EMAIL_CONFIRMATION_PAGE}?${QueryParams.EMAIL_VERIFICATION_TOKEN}=${emailVerificationToken}`;
+    const mailOptions = {
+      to: userEmail,
+      subject: 'Please confirm your email account',
+      html: `
+        Hello,<br> Please Click on the link to verify your email.
```
Comment (karolinayanko) [inline packages/server/src/services/email-sending.service.ts]: I'd suggest to add firstName into email

Comment (dartwlad) [reply packages/server/src/services/email-sending.service.ts]: for now, we don't have such a field. will add in the scope of future PRs.

File: packages/server/src/middlewares/validation.middleware.ts [hunk 2]
```diff
@@ -4,8 +4,8 @@ import { RequestHandler } from 'express';
 import HttpException from '../exceptions/http.exception';
 
 export default function validationMiddleware<T>(
-    // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
-    type,
+  // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
```
Comment (dubbha) [inline packages/server/src/middlewares/validation.middleware.ts]: Turned this off for web,  we may move it to core if you want

File: packages/server/src/models/user.entity.ts [hunk 3]
```diff
@@ -13,4 +13,7 @@ export default class User {
 
   @Column()
   public password: string;
+
+  @Column()
+  public status: string;
```
Comment (hromogen) [inline packages/server/src/models/user.entity.ts:18]: Mb should clarify what status it is?

File: packages/server/src/services/email-sending.service.ts (hunk 1 above)
Comment (hromogen) [reply packages/server/src/services/email-sending.service.ts]: I would also suggest to move text-creator into another entity

File: packages/server/src/models/user.entity.ts (hunk 3 above)
Comment (dartwlad) [reply packages/server/src/models/user.entity.ts:18]: you can check UserStatuses enum in first commit

File: packages/server/src/services/email-sending.service.ts (hunk 1 above)
Comment (karolinayanko) [reply packages/server/src/services/email-sending.service.ts]: Use name for now, and update after a new field will be introduced

**Review by oleksiydubovykTR — APPROVED** [review]

**Review by dubbha — APPROVED** [review]

**Pull request merged**